### PR TITLE
Patch MPI operator installation for RIGs

### PR DESCRIPTION
**Pull request by chnnmz**
# PR Approval Steps

## For Requester

1. Description
    - [ ] Check the PR title and description for clarity. It should describe the changes made and the reason behind them.
    - [ ] Ensure that the PR follows the contribution guidelines, if applicable.

This covers the edge case where customers have HyperPod clusters with 0 standard IGs i.e. RIGs only. In this case, the MPI operator will not deploy. Features like DeepHealthCheck InstanceConnectivity which is based on an NCCL test will not work since that test needs an MPI operator pod. By default, MPI operator pod does not tolerate the taints on RIG workers. Therefore, this is needed to make sure during this RIG-only edge case, MPI operators will still be deployed.

2. Security requirements
    - [ ] Ensure that a Pull Request (PR) does not expose passwords and other sensitive information by using git-secrets and upload relevant evidence: https://github.com/awslabs/git-secrets
    - [ ] Ensure commit has GitHub Commit Signature
3. Manual review
    1. Click on the Files changed tab to see the code changes. Review the changes thoroughly:
        - [ ] Code Quality: Check for coding standards, naming conventions, and readability.
        - [ ] Functionality: Ensure that the changes meet the requirements and that all necessary code paths are tested.
        - [ ] Security: Check for any security issues or vulnerabilities.
        - [ ] Documentation: Confirm that any necessary documentation (code comments, README updates, etc.) has been updated.
4. Check for Merge Conflicts:
    - [ ] Verify if there are any merge conflicts with the base branch. GitHub will usually highlight this. If there are conflicts, you should resolve them.
      
## For Reviewer

1. Go through `For Requester` section to double check each item.
2. Request Changes or Approve the PR:
    1. If the PR is ready to be merged, click Review changes and select Approve.
    2. If changes are required, select Request changes and provide feedback. Be constructive and clear in your feedback.
3. Merging the PR
    1. Check the Merge Method:
        1. Decide on the appropriate merge method based on your repository's guidelines (e.g., Squash and merge, Rebase and merge, or Merge).
    2. Merge the PR:
        1. Click the Merge pull request button.
        2. Confirm the merge by clicking Confirm merge.

### CODE REVIEW

**Comment by haardm** [comment]
LGTM

File: helm_chart/install_rig_dependencies.sh [hunk 1]
```diff
@@ -586,11 +659,11 @@ get_standard_hyperpod_helm_release_name() {
 }
 
 main() {
-    assert_not_already_installed
-
     ensure_yq_installed
     
     set_script_variables
+    
+    assert_not_already_installed
```
[the remark this answers is not in the document]
Comment (chnnmz) [reply helm_chart/install_rig_dependencies.sh:666]: This has already been changed in another PR https://github.com/aws/sagemaker-hyperpod-cli/commit/9fd8d4210f1e069bfcbfad7adb6de4b2702178c8
The variables it uses are actually instantiated in `set_script_variables` so it needs to be run after that

File: helm_chart/install_rig_dependencies.sh [hunk 2]
```diff
@@ -286,6 +288,65 @@ override_training_operators() {
     ]'
 }
 
+override_mpi_operator() {
+    #####################################################
+    # mpi-operator dependency needs to be present
+    # to schedule MPI jobs but by by default, only
```
[the remark this answers is not in the document]
Comment (chnnmz) [reply helm_chart/install_rig_dependencies.sh:294]: Is that what you're referring to?

**Review by emeraldbay — APPROVED** [review]

**Review by zhaoqizqwang — APPROVED** [review]
